core/src/main/java/org/apache/iceberg/actions/FileRewriteGroup.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import java.util.List;
+import org.apache.iceberg.ContentFile;
+import org.apache.iceberg.ContentScanTask;
+
+/**
+ * Container class representing a set of files to be rewritten by a {@link FileRewriteExecutor}.
+ *
+ * @param <I> the Java type of the plan info member like {@link RewriteDataFiles.FileGroupInfo} or
+ *     {@link RewritePositionDeleteFiles.FileGroupInfo}
+ * @param <T> the Java type of the tasks to read the files which are rewritten
+ * @param <F> the Java type of the content files which are rewritten
+ */
+public abstract class FileRewriteGroup<I, T extends ContentScanTask<F>, F extends ContentFile<F>> {
+  private final I info;
+  private final List<T> fileScanTasks;
+  private final long writeMaxFileSize;
+  private final long splitSize;
+  private final int expectedOutputFiles;
+
+  FileRewriteGroup(
+      I info,
+      List<T> fileScanTasks,
+      long writeMaxFileSize,
+      long splitSize,
+      int expectedOutputFiles) {
+    this.info = info;
+    this.fileScanTasks = fileScanTasks;
+    this.writeMaxFileSize = writeMaxFileSize;
+    this.splitSize = splitSize;
+    this.expectedOutputFiles = expectedOutputFiles;
+  }
+
+  /** Identifiers and partition information about the group. */
+  public I info() {
+    return info;
+  }
+
+  /** Input of the group. {@link ContentScanTask}s to read. */
+  public List<T> fileScans() {
```

Comment:
nit: I know the method name is copied from existing code. Since this is a new class, we have an opportunity to name it more accurately as `scanTasks` or `fileScanTasks`.

Reply:
This method exists in the current API as `RewriteFileGroup.fileScans()` and `RewritePositionDeletesGroup.tasks()` and the new method name will be inherited from the `FileRewriteGroup`

Minimally one of them needs to be deprecated, renamed. Do you think we should change both?

Reply:
ok. didn't see the inheritance earlier. probably not worth change right now .